### ISSUE TITLE
CFE_ES_GetAppName() undefined output when failure occurs

### ISSUE THREAD

**Issue by skliper** [nasa/cFE#148]
The `CFE_ES_GetAppName()` function is reasonably well defined - it sets the AppName output buffer and returns `CFE_SUCCESS` when everything went well, and it also nicely ensures that the output buffer is null terminated with the specified output buffer size (all good things).

== PROBLEM: ==
Most callers of this function do not check the return code.

`CFE_ES_GetAppName()` does not touch the output buffer in case of failure (any return code other than `CFE_SUCCESS`) and this would mean that the output may not be null terminated at all, or may be left with garbage from a previous value.

Some callers explicitly add a null char to the buffer at the last position in the buffer, but this still leaves the possibility of treating a buffer full of garbage (or the previous value) as the correct value.

== PROPOSAL ==
As a near term fix `CFE_ES_GetAppName()` should return an empty string in addition to the error code.

**Comment by skliper** [comment]
_Imported from trac issue 117.  Created by jphickey on 2015-10-22T17:28:55, last modified: 2019-03-05T14:58:28_

**Comment by skliper** [comment]
_Trac comment by glimes on 2016-02-16 13:23:48:_

cfe-next is cfe 6.5

**Comment by skliper** [comment]
_Trac comment by glimes on 2016-11-08 14:17:08:_

Current crop of cfe-next are all going into CFE 6.6

**Comment by skliper** [comment]
_Trac comment by jphickey on 2017-10-17 17:22:41:_

Commit [changeset:42025d1] branch `trac-117-getappname-empty-string` proposed as initial fix.  Ready for discussion at least.

I have also reviewed all the places within CFE that `CFE_ES_GetAppName()` is called.  There are about 20 and not a single one actually checks the return code.  Most take the name string and pass it into e.g. CFE_EVS_SendEvent or something similar.

In most cases, if we //were// to fix this to check return code, the most reasonable mitigation would be to fill it with a known string.  The empty string is reasonable, as it will protect against anything really bad happening.

The fact is, that CFE_ES_GetAppName() might be better defined as returning `void` and specifying that in case of failure an empty string is returned, rather than having a return code.  But it doesn't hurt to have both and this keeps compatibility with existing code just in case someone //actually did check the return//.

In other words -- in this case I think keeping the change simple and minimal is the best course of action, and that is what is here.

**Comment by skliper** [comment]
_Trac comment by sstrege on 2017-10-19 13:23:38:_

Recommend/approve - I agree with the simple solution in [changeset:42025d1] and not changing the definition of the API to return void.  In addition I've pushed a supplementary commit with the associated doxygen documentation updates: [changeset:d0d0554]

**Comment by skliper** [comment]
_Trac comment by cdknight on 2017-10-19 17:05:33:_

recommend approve Susie's two commits

(note I feel that ticket #241 may also increase the likelihood of users of the CFE API's checking the return values of calls as we're all kinda used to ignoring int return values ;) .)

**Comment by skliper** [comment]
_Trac comment by acudmore on 2017-10-20 11:28:07:_

recommend approve

**Comment by skliper** [comment]
_Trac comment by glimes on 2017-10-24 15:49:19:_

commit [changeset:42025d1] integrated to development.
commit [changeset:d0d0554] integrated to development.

**Comment by skliper** [comment]
_Trac comment by jhageman on 2019-03-05 14:58:28:_

Milestone renamed
